drivers/pinctrl/pinctrl_ite_it8xxx2.c

```diff
@@ -89,23 +118,110 @@ static int pinctrl_it8xxx2_set(const pinctrl_soc_pin_t *pins)
 	return 0;
 }
 
+static int pinctrl_kscan_it8xxx2_set(const pinctrl_soc_pin_t *pins)
+{
+	const struct pinctrl_it8xxx2_config *pinctrl_config = pins->pinctrls->config;
+	const struct pinctrl_it8xxx2_ksi_kso *ksi_kso = &(pinctrl_config->config.ksi_kso);
+	volatile uint8_t *reg_ctrl = ksi_kso->reg_ctrl;
+	uint8_t pullup_mask = ksi_kso->pullup_mask;
+	uint8_t pp_od_mask = ksi_kso->pp_od_mask;
+	uint32_t pincfg = pins->pincfg;
+
+	/*
+	 * Enable or disable internal pull-up (this bit apply to all pins):
+	 * If KSI[7:0]/KSO[15:0] is in KBS mode , setting 1 enables the internal
+	 * pull-up (KSO[17:16] setting internal pull-up by GPIO port GPCR register).
+	 * If KSI[7:0]/KSO[15:0] is in GPIO mode, then this bit is always disabled.
+	 */
+	switch (IT8XXX2_DT_PINCFG_PULLUP(pincfg)) {
+	case IT8XXX2_PULL_PIN_DEFAULT:
+		/* Disable internal pulll-up */
+		*reg_ctrl &= ~pullup_mask;
+		break;
+	case IT8XXX2_PULL_UP:
+		*reg_ctrl |= pullup_mask;
+		break;
+	default:
+		LOG_ERR("This pull level is not supported.");
+		return -EINVAL;
+	}
+
+	/*
+	 * Set push-pull or open-drain mode (this bit apply to all pins):
+	 * KSI[7:0] doesn't support push-pull and open-drain settings in kbs mode.
+	 * If KSO[17:0] is in KBS mode, setting 1 selects open-drain mode,
+	 * setting 0 selects push-pull mode.
+	 * If KSO[15:0] is in GPIO mode, then this bit is always disabled.
+	 */
+	if (pp_od_mask != NO_FUNC) {
+		switch (IT8XXX2_DT_PINCFG_PP_OD(pincfg)) {
+		case IT8XXX2_PUSH_PULL:
+			*reg_ctrl &= ~pp_od_mask;
+			break;
+		case IT8XXX2_OPEN_DRAIN:
+			*reg_ctrl |= pp_od_mask;
+			break;
+		default:
+			LOG_ERR("This pull mode is not supported.");
+			return -EINVAL;
+		}
+	}
+
+	return 0;
+}
+
+static int pinctrl_kscan_it8xxx2_configure_pins(const pinctrl_soc_pin_t *pins,
+						uint8_t pin_cnt, uint8_t i)
```

Comment:
`pin_cnt` is not used, and you only reference `pins[i]`.  You should just pass `&pins[i]` directly to this function from `pinctcrl_configure_pins()`.  Then you can drop both the `pin_cnt` and 'i' parameters.

Reply:
Done